playbooks/roles/nginx-proxy/templates/nginx-proxy.j2

```diff
@@ -13,6 +13,24 @@ server {
 
     {{ NGINX_PROXY_EXTRA_CONFIG }}
 
+    set $expected_origin false;
+    if ($http_origin ~ {{ NGINX_PROXY_ORIGIN_REGEX }}) {
+    set $expected_origin true;
+    }
+
+    {% for response_code in NGINX_PROXY_CORS_HEADERS_FOR_RESPONSE_CODES %}
+    location @{{ response_code }}_cors_header {
+        if ($expected_origin = true) {
+            add_header 'Access-Control-Allow-Origin' $http_origin always;
+            add_header 'Access-Control-Allow-Methods' 'GET,POST,DELETE,OPTIONS' always;
+            add_header 'Access-Control-Allow-Headers' 'Authorization,DNT,User-Agent,X-Requested-With,If-Modified-Since,Cache-Control,Content-Type,Range' always;
+            add_header 'Access-Control-Expose-Headers' 'Content-Length,Content-Range' always;
+        }
+
+        return {{ response_code }};
```

Comment:
Hmm... I guess that means that for 500 errors nginx will discard the response body. That could be annoying when trying to debug errors. I'm starting to think it might be better to change `NGINX_PROXY_CORS_HEADERS_FOR_RESPONSE_CODES` to only contain `413` for now.